### PR TITLE
[DEVOPS-1120] nix-tools based build

### PR DESCRIPTION
**Pull request by rvl**
Adds a nix-build of cardano-shell which uses the package set defined in `cardano-prelude/snapshot.yaml` instead of a package set from nixpkgs.

This is based on [nix-tools](https://github.com/angerman/nix-tools) rather than `stack2nix`.

It uses the common library code from [iohk-nix](https://github.com/input-output-hk/iohk-nix).

### CODE REVIEW

**Comment by angerman** [comment]
@rvl This seems to build for me now:
```
nix-build nix/pkgs.nix -A cardano-shell.components.exes.cardano-shell-exe -j8  
```

nix-tools needs some work so that `stack-to-nix` produces the proper files. It also need some test aggregation, e.g. build `cardano-shell.components.tests` as well as the tests of dependencies.

File: nix/stack-packages.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{ pkgs ? import <nixpkgs> {} }:
```
Comment (angerman) [inline nix/stack-packages.nix]: As we are doing the overriding via through the module in `pkgs.nix` now, we don't really need this file anymore.

File: stack.yaml [hunk 2]
```diff
@@ -19,7 +19,8 @@
 # resolver: ./custom-snapshot.yaml
 # resolver: https://example.com/snapshots/2018-01-01.yaml
 
-resolver: https://raw.githubusercontent.com/input-output-hk/cardano-prelude/0a32ec92c461e144f981864c97546db11b52e46a/snapshot.yaml
```
Comment (angerman) [inline stack.yaml]: I'll get `nix-tools` to accept this...

File: stack.yaml [hunk 3]
```diff
@@ -36,15 +37,17 @@ resolver: https://raw.githubusercontent.com/input-output-hk/cardano-prelude/0a32
 #  - wai
 packages:
 - .
+- cardano-prelude
+
 # Dependency packages to be pulled from upstream that are not in the resolver
 # using the same syntax as the packages field.
 # (e.g., acme-missiles-0.3)
-extra-deps:
```
Comment (angerman) [inline stack.yaml]: This should have worked, I think? @rvl what issues did you run into?

File: default.nix [hunk 4]
```diff
@@ -0,0 +1,8 @@
+let localLib = import ./nix/lib.nix; in
+{ config ? {}
+, system ? builtins.currentSystem
+# , pkgs ? localLib.importPkgs { inherit config system; }
+, pkgs ? import <nixpkgs> {}
+}:
+
+import ./nix/stack-packages.nix { inherit pkgs; }
```
Comment (angerman) [inline default.nix]:
```suggestion
import ./nix/pkgs.nix { inherit pkgs; }
```

Comment (angerman) [reply default.nix]: See the comment below regarding dropping `stack-packages.nix`

**Comment by angerman** [comment]
```
nix-build . -A cardano-shell.components.exes.cardano-shell-exe -j8
```

seems to produce `./result/bin/cardano-shell-exe`; which prints `Test`.

**Comment by angerman** [comment]
This will now fail, as the test-suite is rigged to fail.

**Comment by rvl** [comment]
Rebased on latest develop.

@ksaric Could you please review? This will change the build and package regeneration process (`README.md` has been updated).

File: shell.nix [hunk 5]
```diff
@@ -1 +1 @@
-(import ./release.nix).cardano-shell.env
+(import ./release.nix).cardano-shell
```
Comment (angerman) [inline shell.nix]: worky worky?

Comment (rvl) [reply shell.nix]: No worky... will put it back to WIP until we can sort this out.

**Comment by CodiePP** [comment]
this nix setup here looks like very much outdated. could this be upgraded to the latest and finest of `nix-tools`, please? @angerman @jbgi

**Comment by ksaric** [comment]
@rvl Closing this one. If somebody needs something from here they can reopen/cherry pick.

**Pull request closed without merging**